src/app/content/utils/seoUtils.ts

```diff
@@ -9,20 +12,182 @@ import {
   splitTitleParts,
 } from './archiveTreeUtils';
 
-const getParentPrefix = (node: LinkedArchiveTreeNode | undefined): string => {
+type PageTypes = 'page' | 'appendix' | 'answer-key' | 'eoc-sub-page' | 'eoc-page' | 'eob-page';
+
+interface DescriptionTemplateValues {
+  parentTitle: string;
+  parentType: 'chapter' | 'book' | 'other';
+  pageTitle: string;
+  bookTitle: string;
+  chapterTitle: string;
+}
+
+const domParser = new DOMParser();
+
+export const getParentPrefix = (node: LinkedArchiveTreeNode | undefined, includeTitle: boolean = false): string => {
   if (!node) {
     return '';
   }
 
   if (archiveTreeSectionIsChapter(node)) {
-    const number = getArchiveTreeSectionNumber(node);
-    return `Ch. ${number} `;
+    const number = getArchiveTreeSectionNumber(node).replace('Chapter', '').trim();
+    const name = getArchiveTreeSectionTitle(node);
+    return includeTitle ? `Ch. ${number}. ${name}` : `Ch. ${number} `;
   }
 
   return archiveTreeSectionIsBook(node.parent)
     ? getArchiveTreeSectionTitle(node) + ' '
-    : getParentPrefix(node.parent);
+    : getParentPrefix(node.parent, includeTitle);
+
+};
+
+const hideMath = (node: Element) => {
+  if (!node) {
+    return '';
+  }
+
+  node.querySelectorAll('.os-math-in-para').forEach((el: Element) => {
+    el.outerHTML = '...';
+  });
+  return node.textContent;
+};
+
+export const getTextContent = (str: string) => {
+  const parsed = domParser.parseFromString(str, 'text/html');
+  const text = parsed.body.textContent;
+  return text;
+};
+
+const removeIntroContent = (node: HTMLElement) => {
+  if (!node) {
+    return null;
+  }
+
+  const introContentList = node.querySelectorAll(
+    '[data-type="abstract"], .learning-objectives, .chapter-objectives, .be-prepared, .os-teacher'
+    ) || [];
+
+  for (let i = 0; i <= introContentList.length; i++) {
+    if (introContentList[i]) {
+      const parent = introContentList[i].parentNode;
```

Comment:
is there not a remove self or something? https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove

is there a reason you're not using that?

Reply:
you're right - I may have been worried about IE compatibility but I see it's used elsewhere